src/jti_content.js

```diff
@@ -1,169 +1,256 @@
+                    // Found closing Tag!
+                    selectEnd = (i + 5);
+                    break;
+                }
+            }
+            if (selectEnd) {
+                // Select all the text between the two tags.
+                $(this)[0].setSelectionRange(selectStart, selectEnd);
+                cursorStart = cursorFinish = selectStart; // Set the cursor position to the select start point. This will ensure we find the next <TI> tag when using keyboard shortcut
+            } else { // This only happens when user clicks on the the closing <TI> tag. Set selectStart to null so that it wont break the keyborad functionality
+                selectStart = null;
+            }
+        }
+    }
+
+    // Detect ctrl or cmd pressed
+    $('#description').keydown(function (e) {
+        if (e.keyCode === getKeyCodeOfCtrlOrCmd()) ctrlDown = true;
+    }).keyup(function (e) {
+        if (e.keyCode === getKeyCodeOfCtrlOrCmd()) ctrlDown = false;
+    });
+
+    // Keypress listener
+    $('#description').keydown(function (e) {
+        if (ctrlDown && (e.keyCode === backtickKey)) { // If ctrl is pressed
+            let {start: tagStartIndex, end: tagEndIndex} = getAllIndexes($(this).val()); // Find all <TI> and </TI> tags in selected template.
+            if (tagStartIndex.length !== 0 && tagEndIndex.length !== 0) { // Works only if the selected template contains any <TI> tag
+                if (selectStart === null && selectEnd === null) { // Start from first <TI>
+                    var StartPos = FindNextTI(cursorStart, tagStartIndex, tagEndIndex); // Find the starting <TI> tag
+                    $(this)[0].setSelectionRange(StartPos.start, StartPos.end);
+                    selectStart = StartPos.start; // Set Start Index
+                    selectEnd = StartPos.end; // Set End Index
+                } else { // Select next <TI> set
+                    if (tagStartIndex.indexOf(selectStart) === tagStartIndex.length - 1 && tagEndIndex.indexOf(selectEnd) === tagEndIndex.length - 1) { // Currently selecting the last set of <TI>, back to first set
+                        $(this)[0].setSelectionRange(tagStartIndex[0], tagEndIndex[0]);
+                        selectStart = tagStartIndex[0];
+                        selectEnd = tagEndIndex[0];
+                    } else {
+                        if (tagStartIndex.indexOf(selectStart) === -1 && tagEndIndex.indexOf(selectEnd) === -1) { // Highlighted <TI> tag is modified by user. Now we need search for the next <TI>.
+                            if (cursorStart < selectStart) cursorStart = selectStart;
+                            StartPos = FindNextTI(cursorStart, tagStartIndex, tagEndIndex); // Find the starting <TI> tag
+                            $(this)[0].setSelectionRange(StartPos.start, StartPos.end);
+                            selectStart = StartPos.start; // Set Start Index
+                            selectEnd = StartPos.end; // Set End Index

```

Comment:
could possibly wrap this step of setting selection range, start index, and end index in a local function since you do it twice